### PR TITLE
Sr25519Signer takes a private key instead of a seed

### PR DESCRIPTION
**Pull request by kevin-wu-cere**
Sr25519Signer takes a private key instead of a seed so that it is consistent with other signing schemes.

### CODE REVIEW

**Comment by kevin-wu-cere** [comment]
I think most likely some test cases are failing because of https://github.com/Cerebellum-Network/ddc-node/pull/183: maybe we could think about disabling that check for end-to-end tests outside ddc-node.

**Comment by upalinski** [comment]
@kevin-wu-cere i think we can easily avoid scaling by increasing mocked application limits in end to end tests

**Review by upalinski — APPROVED** [review]

**Comment by upalinski** [comment]
@kevin-wu-cere actually in combination with this pull request https://github.com/Cerebellum-Network/ddc-node/pull/189/files app will not scale until tier is update in smart contract (what isn't possible with mocked smart contract response that is configured via env variables).

For now i suggest to disable producer and consumers tests that are testing behaviour during application scaling (other tests should be active).

**Review by svalasovich — APPROVED** [review]
LGTM, I think we can turn off some test and keep only general scenario.

File: src/test/kotlin/network/cere/ddc/client/consumer/DdcConsumerTest.kt [hunk 1]
```diff
@@ -1170,6 +1170,6 @@ internal class DdcConsumerTest {
     private fun assertPieces(expected: Collection<Piece>, actual: Collection<Piece>) {
         assertEquals(expected.size, actual.size, "Different number of pieces")
 
-        assertEquals(expected.toSet(), actual.map { it.copy(checksum = null) }.toSet())
+        assertEquals(expected.map { it.copy(offset = null) }.toSet(), actual.map { it.copy(checksum = null, offset = null) }.toSet())
```
Comment (kevin-wu-cere) [inline src/test/kotlin/network/cere/ddc/client/consumer/DdcConsumerTest.kt:1173]: @upalinski @svalasovich I've disabled the app scaling related tests and the "offset" checking: this change changes partition allocation (so piece assignment) https://github.com/Cerebellum-Network/ddc-node/commit/62766eb85c1a1dcd9eba982c70e782f41ae60edf#diff-d86e827d04ab1261d84b64a263a42e0b88816fc24f8362db1e86e6c79736467eR351. As a ddc-client, it probably doesn't need to care about the internals (which partition the data was saved on, etc). Let me know if you have any concerns.

Comment (upalinski) [reply src/test/kotlin/network/cere/ddc/client/consumer/DdcConsumerTest.kt:1173]: ok

**Pull request merged**